src/sync-merkleroots.test.ts

```diff
@@ -0,0 +1,315 @@
+];
+
+// mockedAPIResponseFn is a mock of SPV-Wallet it will return a paged response of merkle roots since last evaluated merkle root
+const mockedAPIResponseFn = (lastMerkleRoot: string): ExclusiveStartKeyPage<MerkleRoot[]> => {
+  if (lastMerkleRoot === '') {
+    return {
+      content: mockedSPVWalletData,
+      page: {
+        lastEvaluatedKey: '',
+        totalElements: mockedSPVWalletData.length,
+        size: mockedSPVWalletData.length,
+      },
+    };
+  }
+
+  const lastMerkleRootIdx = mockedSPVWalletData.findIndex((v) => v.merkleRoot === lastMerkleRoot);
+
+  // handle case when lastMerkleRoot is already highest in the servers database
+  if (lastMerkleRootIdx === mockedSPVWalletData.length - 1) {
+    return {
+      content: [],
+      page: {
+        lastEvaluatedKey: '',
+        totalElements: mockedSPVWalletData.length,
+        size: 0,
+      },
+    };
+  }
+
+  const content = mockedSPVWalletData.slice(lastMerkleRootIdx + 1);
+
+  return {
+    content,
+    page: {
+      lastEvaluatedKey: content[content.length - 1].merkleRoot,
+      totalElements: mockedSPVWalletData.length,
+      size: content.length,
+    },
+  };
+};
+
+beforeAll(() => {
+  fetchMock.enableMocks();
+});
+
+beforeEach(() => {
+  fetchMock.resetMocks();
+});
+
+afterAll(() => {
+  fetchMock.disableMocks();
+});
+
+describe('Test sync merkle roots success scenarios', () => {
+  test('Should properly sync database when empty', async () => {
+    // setup
+    fetchMock.mockResponse(async (req) => {
+      const url = new URL(req.url);
+      const queryParams = url.searchParams;
+
+      const lastEvaluatedKey = queryParams.get('lastEvaluatedKey') || '';
+
+      return Promise.resolve({
+        body: JSON.stringify(mockedAPIResponseFn(lastEvaluatedKey)),
+        headers: { 'Content-Type': 'application/json' },
+      });
+    });
+
+    // given
+    const clientDb: {
+      merkleRoots: MerkleRoot[];
+    } = {
+      merkleRoots: [],
+    };
+    const repository = createRepository(clientDb);
+
+    // when
+    await client.SyncMerkleRoots(repository);
+
+    //then
+    expect(clientDb.merkleRoots.length).toBe(mockedSPVWalletData.length);
+    expect(clientDb.merkleRoots[clientDb.merkleRoots.length - 1].merkleRoot).toBe(
+      mockedSPVWalletData[mockedSPVWalletData.length - 1].merkleRoot,
+    );
+    expect(clientDb.merkleRoots[clientDb.merkleRoots.length - 1].blockHeight).toBe(
+      mockedSPVWalletData[mockedSPVWalletData.length - 1].blockHeight,
+    );
+  });
+
+  test('Should properly sync database when partially filled', async () => {
+    // setup
+    fetchMock.mockResponse(async (req) => {
+      const url = new URL(req.url);
+      const queryParams = url.searchParams;
+
+      const lastEvaluatedKey = queryParams.get('lastEvaluatedKey') || '';
+
+      return Promise.resolve({
+        body: JSON.stringify(mockedAPIResponseFn(lastEvaluatedKey)),
+        headers: { 'Content-Type': 'application/json' },
+      });
+    });
```

Comment:
I can see that all three `mockResponse` definitions look almost the same. Consider move the logic to a function and use it 😄